### PR TITLE
AO3-6545 Default FAQ to last valid locale instead of always English

### PR DESCRIPTION
**Pull request by ceithir**
# Pull Request Checklist

* [x] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [x] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [x] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [x] Have you added the [Jira](https://otwarchive.atlassian.net) issue number as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [x] Do you have fewer than 5 pull requests already open? If not, please wait until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6545

## Purpose

If accessing FAQ with no specified locale or an invalid locale, redirect to last locale in session, then user preferred locale, then default locale.

## Testing Instructions

As per JIRA. To be tested as guest, logged in standard user, and admin.

## Credit

Ceithir (he/him)

### CODE REVIEW

File: spec/controllers/archive_faqs_controller_spec.rb [hunk 1]
```diff
@@ -6,38 +6,43 @@
...
+        expect(I18n).not_to receive(:with_locale)
+        get :index, params: { language_id: "" }
+        it_redirects_to(archive_faqs_path(language_id: I18n.default_locale))
+      end
 
-    it "redirects to the default locale when the locale param and the session locale are empty" do
-      expect(I18n).not_to receive(:with_locale)
-      get :index, params: { language_id: "" }, session: { language_id: "" }
-      it_redirects_to(archive_faqs_path(language_id: "en"))
+      it "redirects to the default locale when the locale param and the session locale are empty" do
```
Comment (brianjaustin) [inline spec/controllers/archive_faqs_controller_spec.rb:32]: Isn't this redundant with `redirects to the default locale when the locale param is empty`, since the context is that `session[:locale]` is unset?

File: spec/controllers/archive_faqs_controller_spec.rb [hunk 2]
```diff
@@ -6,38 +6,43 @@
...
-      expect(I18n).to receive(:with_locale).with("eldritch").and_call_original
-      get :index, params: { language_id: "eldritch" }
-      expect(response).to render_template(:index)
-    end
+  let(:non_standard_locale) { create(:locale) }
+  let(:user_locale) { create(:locale) }
+  let(:user) do
+    user = create(:user)
+    user.preference.update!(preferred_locale: user_locale.id)
+    $rollout.activate_user(:set_locale_preference, user)
```
Comment (brianjaustin) [inline spec/controllers/archive_faqs_controller_spec.rb:14]: I'm not sure if this is on for everyone in prod, but it might be nice to test when it's disabled as well

File: spec/controllers/archive_faqs_controller_spec.rb (hunk 1 above)
Comment (ceithir) [reply spec/controllers/archive_faqs_controller_spec.rb:32]: This test, as is, is useful to catch an edge case forcing to add a couple of `.presence`, the one where `session[:language_id]` is blank (nil actually), which isn't something possible with the new code, but used to be with the old one. So it's kinda a legacy failsafe.

Name should definitely be updated to reflect that though.

File: spec/controllers/archive_faqs_controller_spec.rb [hunk 3]
```diff
@@ -39,10 +38,20 @@
     context "when logged in as a regular user" do
       before { fake_login_known_user(user) }
 
-      it "redirects to the user preferred locale when the locale param is invalid" do
+      it "redirects to the default locale when the locale param is invalid" do
```
Comment (brianjaustin) [inline spec/controllers/archive_faqs_controller_spec.rb:41]: I think it might make understanding this easier if it's wrapped in a context as well. Something like
```rb
context "when the set locale preference feature flag is off" do
```

File: spec/controllers/archive_faqs_controller_spec.rb [hunk 4]
```diff
@@ -39,10 +38,20 @@
...
         expect(I18n).not_to receive(:with_locale)
         get :index, params: { language_id: "eldritch" }
-        it_redirects_to(archive_faqs_path(language_id: user_locale.iso))
+        it_redirects_to(archive_faqs_path(language_id: I18n.default_locale))
+      end
+
+      context "with set_locale_preference" do
```
Comment (brianjaustin) [inline spec/controllers/archive_faqs_controller_spec.rb:47]: Similar to above,
```suggestion
      context "when the set_locale_preference feature flag is on" do
```

File: spec/controllers/archive_faqs_controller_spec.rb (hunk 1 above)
Comment (brianjaustin) [reply spec/controllers/archive_faqs_controller_spec.rb]: Ah, gotcha. Feel free to mark this thread as resolved too

**Review by brianjaustin — COMMENTED** [review]
Thanks for the added test! Just a couple more wording suggestions, otherwise looks good to me

**Review by brianjaustin — APPROVED** [review]
